src/uu/dd/src/progress.rs

```diff
@@ -51,6 +51,76 @@ pub(crate) struct ProgUpdate {
     pub(crate) complete: bool,
 }
 
+/// Get the precision to use for displaying a given time duration.
+///
+/// This function returns a number of digits to display after the
+/// decimal. It is calibrated to match the behavior of GNU `dd`. For
+/// example, here are some samples of durations presented by `dd`:
+///
+/// ```text
+/// 1566.68 s
+///  144.31 s
+///   14.4933 s
+///    1.5546 s
+///    0.251168 s
+///    0.0485443 s
+///    0.00701855 s
+///    0.000415484 s
+/// ```
```

Comment:
I just tested %g in C and got these formats, which is way more logical, because it's just 6 significant figures. However, it's trimming trailing zeros, which explains the formats you got.
```
1.23456e+02
1.23456e+01
123456
 12345.6
  1234.56
   123.456
    12.3456
     1.23456
     0.123456
     0.0123456
     0.00123456
     0.000123456
```